File: slack-reader/src/test/java/dev/number6/slackreader/generate/ChannelHistoryGenerator.java
package dev.number6.slackreader.generate;


import dev.number6.slackreader.model.ChannelHistoryResponse;
import dev.number6.slackreader.model.Message;
import uk.org.fyodor.generators.Generator;

import java.util.List;

class ChannelHistoryGenerator implements Generator<ChannelHistoryResponse> {
    private final Generator<Boolean> booleanGenerator = SlackReaderRDG.bool();
    private final Generator<Long> timestampGenerator = SlackReaderRDG.longVal();
    private final Generator<List<Message>> messageListGenerator = SlackReaderRDG.list(SlackReaderRDG.message());

    @Override
    public ChannelHistoryResponse next() {

        return new ChannelHistoryResponse(booleanGenerator.next(),
                messageListGenerator.next(),
                booleanGenerator.next(),
                timestampGenerator.next(),
                timestampGenerator.next());
    }
}

File: slack-reader/src/test/kotlin/dev/number6/slackreader/generate/ChannelHistoryGenerator.kt
package dev.number6.slackreader.generate

import dev.number6.slack.model.ChannelHistoryResponse
import uk.org.fyodor.generators.Generator
import uk.org.fyodor.generators.RDG

internal class ChannelHistoryGenerator : Generator<ChannelHistoryResponse> {
    private val booleanGenerator = RDG.bool()
    private val timestampGenerator = RDG.longVal()
    private val messageListGenerator = RDG.list(SlackReaderRDG.message())
    override fun next(): ChannelHistoryResponse {
        return ChannelHistoryResponse(booleanGenerator.next(),
                messageListGenerator.next(),
                booleanGenerator.next(),
                timestampGenerator.next(),
                timestampGenerator.next())
    }
}